<!DOCTYPE html>

<html lang="en-US" xml:lang="en-US">

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />


<meta name="date" content="2022-06-08" />

<title>Case study: colocalization analysis in prostate cancer</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">Case study: colocalization analysis in prostate cancer</h1>
<h4 class="date">2022-06-08</h4>



<div id="before-we-start." class="section level2">
<h2>Before we start.</h2>
<p>All we need to prepare include three parts:</p>
<ol style="list-style-type: decimal">
<li>GWAS summary statistics dataset.</li>
<li>Name of tissue of interest.</li>
<li>Following three installed packages.</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(xQTLbiolinks)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span></code></pre></div>
<p>Prostate cancer is one of the most common cancers in men. Prostate cancer pathogenesis involves both heritable and environmental factors. The molecular events involved in the development or progression of prostate cancer are still unclear. In this example, we aim to figure out the causal variants and genes assocaited with prostate cancer, and to uncover potential molecular mechanisms of regulation.</p>
<p>For data preparation, we download summary statistics dataset of a GWAS study (<a href="http://ftp.ebi.ac.uk/pub/databases/gwas/summary_statistics/GCST006001-GCST007000/GCST006085/harmonised/29892016-GCST006085-EFO_0001663-build37.f.tsv.gz">GCST006085</a>) of prostate cancer from GWAS category and load the dataset in R with <code>data.table</code> package. Correspondingly, we chose tissue <code>Porstate</code> for study. We retain the variants with dbSNP id (start with <code>rs</code>), and a data.table object named <code>gwasDF</code> of 13,498,990 (rows) x 5 (cols) is loaded.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>gwasDF <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;29892016-GCST006085-EFO_0001663-build37.f.tsv.gz&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># extract columns.</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>gwasDF<span class="ot">&lt;-</span> gwasDF[<span class="fu">str_detect</span>(variant_id, <span class="st">&quot;^rs&quot;</span>),.(<span class="at">rsid=</span>variant_id, <span class="at">chrom=</span>chromosome, <span class="at">position=</span> base_pair_location, <span class="at">pValue=</span>p_value, <span class="at">AF=</span>effect_allele_frequency)]</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># tissue:</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>tissueSiteDetail<span class="ot">=</span><span class="st">&quot;Prostate&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gwasDF)</span></code></pre></div>
<pre><code>#&gt;     chr position chr_position        rsid    beta     se      N p-value    maf
#&gt; 1: chr1    10177   chr1:10177 rs201752861 -0.0211 0.0136 140254  0.1217 0.6104
#&gt; 2: chr1    11008   chr1:11008 rs575272151 -0.0204 0.0196 140254  0.2979 0.9058
#&gt; 3: chr1    11012   chr1:11012 rs544419019 -0.0204 0.0196 140254  0.2979 0.9058
#&gt; 4: chr1    13110   chr1:13110 rs540538026 -0.0377 0.0332 140254  0.2565 0.0589
#&gt; 5: chr1    13116   chr1:13116  rs62635286 -0.0035 0.0178 140254  0.8447 0.8255
#&gt; 6: chr1    13118   chr1:13118 rs200579949 -0.0035 0.0178 140254  0.8447 0.8255</code></pre>
</div>
<div id="step-1-filter-sentinel-snps." class="section level2">
<h2>Step 1: Filter sentinel snps.</h2>
<p>Sentinel SNP is the most prominent signal within a given genome range, and is usually in high LD with causal variants. By default, xQTLbiolinks detect sentinel snps that with the p-value &lt; 5e-8 and SNP-to-SNP distance &gt; 10e6 bp. <strong>Note</strong>: For in this example, due to the inconsistent genome version between the GWAS dataset (GRCh37) and eQTL associations (GRCh38) from eQTL category, conversion of genome version is required, and can be conducted using <code>xQTLanalyze_getSentinelSnp</code> with <code>genomeVersion=&quot;grch37&quot;</code> and <code>grch37To38=TRUE</code> (package <code>rtracklayer</code>is required):</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>sentinelSnpDF <span class="ot">&lt;-</span> <span class="fu">xQTLanalyze_getSentinelSnp</span>(gwasDF, <span class="at">centerRange=</span><span class="fl">1e6</span>,</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">genomeVersion=</span><span class="st">&quot;grch37&quot;</span>, <span class="at">grch37To38=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>A total of 94 sentinel SNPs are detected.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sentinelSnpDF)</span></code></pre></div>
<pre><code>#&gt;           rsid  chr  position    pValue    maf
#&gt; 1:  rs55664108 chr1 204587862 8.572e-25 0.7157
#&gt; 2:  rs35296356 chr1 150601662 3.355e-14 0.3384
#&gt; 3:  rs34579442 chr1 153927424 4.478e-14 0.6637
#&gt; 4: rs146564277 chr1 155051619 1.577e-12 0.0301
#&gt; 5:  rs34848415 chr1 205762038 2.893e-09 0.5231
#&gt; 6:  rs56391074 chr1  87745032 1.659e-08 0.6298</code></pre>
</div>
<div id="step-2-identify-trait-genes-for-each-sentinel-snps." class="section level2">
<h2>Step 2: Identify trait genes for each sentinel SNPs.</h2>
<p>Trait genes are genes that located in the range of 1Mb (default, can be changed with parameter <code>detectRange</code>) of sentinel SNPs. In order to reduce the number of trait genes and thus reduce the running time, we take the overlap of eGenes and trait genes as the final output of the function <code>xQTLanalyze_getTraits</code>:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>traitsAll <span class="ot">&lt;-</span> <span class="fu">xQTLanalyze_getTraits</span>(sentinelSnpDF, <span class="at">detectRange=</span><span class="fl">1e6</span>, <span class="at">tissueSiteDetail=</span>tissueSiteDetail)</span></code></pre></div>
<p>Totally, 898 associations between 835 traits genes and 92 sentinel SNPs are detected</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(example_Coloc_traitsAll)</span></code></pre></div>
<pre><code>#&gt;    chromosome geneStart   geneEnd geneStrand geneSymbol          gencodeId
#&gt; 1:       chr1 205086142 205122015          -      RBBP5 ENSG00000117222.13
#&gt; 2:       chr1 205142505 205211566          -      DSTYK ENSG00000133059.16
#&gt; 3:       chr1 205336065 205357090          -    KLHDC8A ENSG00000162873.14
#&gt; 4:       chr1 204828651 205022822          +      NFASC ENSG00000163531.15
#&gt; 5:       chr1 204190341 204196486          -      KISS1  ENSG00000170498.8
#&gt; 6:       chr1 204198160 204214092          -     GOLT1A  ENSG00000174567.7
#&gt;          rsid  position    pValue    maf
#&gt; 1: rs55664108 204587862 8.572e-25 0.7157
#&gt; 2: rs55664108 204587862 8.572e-25 0.7157
#&gt; 3: rs55664108 204587862 8.572e-25 0.7157
#&gt; 4: rs55664108 204587862 8.572e-25 0.7157
#&gt; 5: rs55664108 204587862 8.572e-25 0.7157
#&gt; 6: rs55664108 204587862 8.572e-25 0.7157</code></pre>
</div>
<div id="step-3-conduct-colocalization-analysis." class="section level2">
<h2>Step 3: Conduct colocalization analysis.</h2>
<p>Following three steps of colocalization analysis are encapsulated in one function xQTLanalyze_coloc:</p>
<ol style="list-style-type: decimal">
<li>Retrieved all associations from EBI eQTL catalogue for a specified gene.</li>
<li>Merge the data.frame of GWAS and eQTL by rsid.</li>
<li>Perform colocalization analysis using coloc package.</li>
</ol>
<p>For above 835 trait genes, a for loop can be used to get these genes’ outputs of colocalization analysis (this may take several hours):</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>genesAll<span class="ot">&lt;-</span> <span class="fu">unique</span>(traitsAll<span class="sc">$</span>gencodeId)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>colocResultAll <span class="ot">&lt;-</span> <span class="fu">data.table</span>()</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(genesAll)){</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  colocResult <span class="ot">&lt;-</span> <span class="fu">xQTLanalyze_coloc</span>(gwasDF, </span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">genomeVersion =</span> <span class="st">&quot;grch37&quot;</span>, </span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>                                   genesAll[i], <span class="at">tissueSiteDetail=</span>tissueSiteDetail)<span class="sc">$</span>coloc_Out_summary</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(colocResult)){ colocResultAll <span class="ot">&lt;-</span> <span class="fu">rbind</span>(colocResultAll, colocResult)}</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">message</span>(<span class="fu">format</span>(<span class="fu">Sys.time</span>(), <span class="st">&quot;== %Y-%b-%d %H:%M:%S &quot;</span>),<span class="st">&quot; == Id:&quot;</span>,i,<span class="st">&quot;/&quot;</span>,<span class="fu">length</span>(genesAll),<span class="st">&quot; == Gene:&quot;</span>,genesAll[i])</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>In this case, we invoke the funciton <code>coloc.abf</code> from package <code>coloc</code> to estimate the posterior support of variants for each hypothesis: H0,H1,H2,H3,H4:</p>
<ul>
<li>H0 : neither trait has a genetic association in the region</li>
<li>H1: only trait 1 has a genetic association in the region</li>
<li>H2: only trait 2 has a genetic association in the region</li>
<li>H3: both traits are associated, but with different causal variants</li>
<li>H4: both traits are associated and share a single causal variant</li>
</ul>
<p>Output is a data.table that combined all results of <code>coloc_Out_summary</code> of all genes. The posterior probability of variants to each hypothesis (H0-H4) are listed in <code>colocResultAll</code>:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(colocResultAll)</span></code></pre></div>
<pre><code>#&gt;    nsnps    PP.H0.abf   PP.H1.abf    PP.H2.abf PP.H3.abf    PP.H4.abf
#&gt; 1:  5666 3.339109e-19 0.333928806 6.435451e-19 0.6435572 2.251401e-02
#&gt; 2:  5509 6.902252e-20 0.069026225 9.265196e-19 0.9265649 4.408886e-03
#&gt; 3:  5072 6.447026e-19 0.644737236 3.172030e-19 0.3171819 3.808084e-02
#&gt; 4:  5863 4.318128e-22 0.000431836 9.994821e-19 0.9995357 3.245264e-05
#&gt; 5:  5688 5.603139e-19 0.560343948 3.653347e-19 0.3652799 7.437616e-02
#&gt; 6:  5661 3.421306e-19 0.342148992 6.147008e-19 0.6146906 4.316040e-02
#&gt;             traitGene
#&gt; 1: ENSG00000117222.13
#&gt; 2: ENSG00000133059.16
#&gt; 3: ENSG00000162873.14
#&gt; 4: ENSG00000163531.15
#&gt; 5:  ENSG00000170498.8
#&gt; 6:  ENSG00000174567.7</code></pre>
<p>To save time and go through this case as soon as possible, you can get the above result directly with:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>colocResultAll <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;https://raw.githubusercontent.com/dingruofan/exampleData/master/colocResultAll.txt&quot;</span>)</span></code></pre></div>
</div>
<div id="step-4-visualization-of-the-results." class="section level2">
<h2>Step 4: Visualization of the results.</h2>
<p>We considered colocalization tests with a posterior probability of hypothesis 4 (PPH4.ABF) ≥ 0.75 as having strong or moderate evidence for colocalization.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>colocResultsig <span class="ot">&lt;-</span> colocResultAll[PP.H4.abf<span class="sc">&gt;</span><span class="fl">0.75</span>][<span class="fu">order</span>(<span class="sc">-</span>PP.H4.abf)]</span></code></pre></div>
<p>There are 27 trait genes that are associated and share a single causal variant:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(colocResultsig)</span></code></pre></div>
<pre><code>#&gt;    nsnps    PP.H0.abf    PP.H1.abf    PP.H2.abf   PP.H3.abf PP.H4.abf
#&gt; 1:  7108 3.722120e-16 7.637674e-04 1.159909e-15 0.001382244 0.9978540
#&gt; 2:  5034 2.405778e-32 1.349362e-05 8.271165e-30 0.003642818 0.9963437
#&gt; 3:  6045 1.551367e-07 1.220850e-05 1.078421e-04 0.007494261 0.9923855
#&gt; 4:  5389 1.594413e-24 5.835862e-07 2.875803e-20 0.009535535 0.9904639
#&gt; 5:  6154 4.387457e-18 6.631626e-03 6.233624e-18 0.008437170 0.9849312
#&gt; 6:  4616 1.043225e-19 1.214959e-02 6.298351e-20 0.006353677 0.9814967
#&gt;             traitGene
#&gt; 1:  ENSG00000137673.8
#&gt; 2: ENSG00000167641.10
#&gt; 3: ENSG00000184058.12
#&gt; 4: ENSG00000115486.11
#&gt; 5: ENSG00000179409.10
#&gt; 6:  ENSG00000277744.1</code></pre>
<p>All these genes’ details can be fetched with <code>xQTLquery_gene</code>:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>outGenes <span class="ot">&lt;-</span> <span class="fu">xQTLquery_gene</span>(colocResultsig<span class="sc">$</span>traitGene)</span></code></pre></div>
<p>Add the value of PPH4 for each gene, and remove non-protein-coding genes:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>outGenes <span class="ot">&lt;-</span> <span class="fu">merge</span>(colocResultsig[,.(<span class="at">gencodeId=</span> traitGene, PP.H4.abf)], </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>                  outGenes[,.(geneSymbol, gencodeId, entrezGeneId, geneType)], <span class="at">by=</span><span class="st">&quot;gencodeId&quot;</span>, <span class="at">sort=</span><span class="cn">FALSE</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>outGenes <span class="ot">&lt;-</span> outGenes[geneType <span class="sc">==</span><span class="st">&quot;protein coding&quot;</span>]</span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>outGenes</span></code></pre></div>
<pre><code>#&gt;              gencodeId PP.H4.abf geneSymbol entrezGeneId       geneType
#&gt;  1:  ENSG00000137673.8 0.9978540       MMP7         4316 protein coding
#&gt;  2: ENSG00000167641.10 0.9963437   PPP1R14A        94274 protein coding
#&gt;  3: ENSG00000184058.12 0.9923855       TBX1         6899 protein coding
#&gt;  4: ENSG00000115486.11 0.9904639       GGCX         2677 protein coding
#&gt;  5: ENSG00000179409.10 0.9849312     GEMIN4        50628 protein coding
#&gt;  6: ENSG00000117280.12 0.9701717      RAB29         8934 protein coding
#&gt;  7: ENSG00000184012.11 0.9652282    TMPRSS2         7113 protein coding
#&gt;  8: ENSG00000069275.12 0.9625282     NUCKS1        64710 protein coding
#&gt;  9: ENSG00000099331.13 0.9607259      MYO9B         4650 protein coding
#&gt; 10: ENSG00000155749.12 0.9538274   ALS2CR12       130540 protein coding
#&gt; 11: ENSG00000118961.14 0.9294589       LDAH        60526 protein coding
#&gt; 12: ENSG00000101751.10 0.9271138       POLI        11201 protein coding
#&gt; 13: ENSG00000167695.14 0.9203321     FAM57A        79850 protein coding
#&gt; 14:  ENSG00000172613.7 0.9099755      RAD9A         5883 protein coding
#&gt; 15:  ENSG00000180535.3 0.8941134    BHLHA15       168620 protein coding
#&gt; 16: ENSG00000003400.14 0.8932643     CASP10          843 protein coding
#&gt; 17: ENSG00000115648.13 0.8889021       MLPH        79083 protein coding
#&gt; 18: ENSG00000162877.12 0.8849219     PM20D1       148811 protein coding
#&gt; 19: ENSG00000204536.13 0.8733027     CCHCR1        54535 protein coding
#&gt; 20:  ENSG00000091844.7 0.8629044      RGS17        26575 protein coding
#&gt; 21:  ENSG00000083937.8 0.8388545     CHMP2B        25978 protein coding
#&gt; 22: ENSG00000136819.15 0.8365468    C9orf78        51759 protein coding
#&gt; 23: ENSG00000065060.16 0.7937810   UHRF1BP1        54887 protein coding
#&gt; 24: ENSG00000198625.12 0.7695907       MDM4         4194 protein coding
#&gt;              gencodeId PP.H4.abf geneSymbol entrezGeneId       geneType</code></pre>
<p>Ridgeline plot can be used to compare the expressions among these genes:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">xQTLvisual_genesExp</span>(outGenes<span class="sc">$</span>geneSymbol, <span class="at">tissueSiteDetail=</span>tissueSiteDetail)</span></code></pre></div>
<p>Trait gene <strong><code>MMP7</code></strong> that with the highest <code>PPH4.ABF=0.9978</code> encodes a member of the peptidase M10 family of matrix metalloproteinases, which is involved in the breakdown of extracellular matrix in normal physiological processes, such as embryonic development, reproduction, and tissue remodeling, as well as in disease processes, such as arthritis and metastasis <span class="citation">(<a href="#ref-RefSeq" role="doc-biblioref"><span>“RefSeq,”</span> n.d.</a>)</span>. Prostate cancer can be promoted via MMP7-induced epithelial-to-mesenchymal transition by Interleukin-17 <span class="citation">(<a href="#ref-zhang2017" role="doc-biblioref">Zhang et al. 2017</a>)</span>. Resent literature has shown that serum MMP7 levels could guide metastatic therapy for prostate cancer <span class="citation">(<a href="#ref-tregunna2020" role="doc-biblioref">Tregunna 2020</a>)</span>.</p>
<p>Expression of MMP7 in multiple tissues can be plotted with <code>xQTLvisual_geneExpTissues</code>:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>geneExpTissues <span class="ot">&lt;-</span> <span class="fu">xQTLvisual_geneExpTissues</span>(<span class="st">&quot;MMP7&quot;</span>, <span class="at">log10y =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>The number and significance of eQTLs in distinguished tissues are capable of showing a tissue-specific effect or a ubiquitous effect. The function <code>xQTLvisual_eqtl</code> can be used to indicate whether the gene is widely regulation in various tissues.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">xQTLvisual_eqtl</span>(<span class="st">&quot;MMP7&quot;</span>)</span></code></pre></div>
<p>Besides, we provide functions <code>xQTLvisual_locusCompare</code> and <code>xQTLvisual_locusZoom</code> to visualize the colocalization between the GWAS and the eQTL dataset for a specified gene, we take the gene <code>MMP7</code> as an example:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Download all eQTL associations of gene  MMP7 in prostate:</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>eqtlAsso <span class="ot">&lt;-</span> <span class="fu">xQTLdownload_eqtlAllAsso</span>(<span class="at">gene=</span><span class="st">&quot;MMP7&quot;</span>,<span class="at">tissueLabel =</span> tissueSiteDetail)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge the variants of GWAS and eQTL dataset by rsid:</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>gwasEqtldata <span class="ot">&lt;-</span> <span class="fu">merge</span>(gwasDF[,<span class="sc">-</span><span class="fu">c</span>(<span class="st">&quot;AF&quot;</span>)], eqtlAsso[,.(<span class="at">rsid=</span>snpId, pValue)],</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;rsid&quot;</span>), <span class="at">suffixes =</span> <span class="fu">c</span>(<span class="st">&quot;.gwas&quot;</span>,<span class="st">&quot;.eqtl&quot;</span>))</span></code></pre></div>
<p>Five retained fields are required:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>gwasEqtldata</span></code></pre></div>
<pre><code>#&gt;             rsid chrom position pValue.gwas pValue.eqtl
#&gt;    1: rs10000104  chr4 10190142      0.2830   0.4986640
#&gt;    2: rs10000248  chr4 10720872      0.0359   0.6089460
#&gt;    3: rs10000318  chr4 11613397      0.5810   0.3734920
#&gt;    4: rs10000369  chr4 11234631      0.0453   0.6642180
#&gt;    5: rs10000399  chr4 11595842      0.2370   0.0948472
#&gt;   ---                                                  
#&gt; 8007:  rs9999345  chr4 10895379      0.0861   0.0100352
#&gt; 8008:  rs9999470  chr4 10315399      0.2300   0.0700527
#&gt; 8009:  rs9999523  chr4 11410463      0.9930   0.2258890
#&gt; 8010:  rs9999669  chr4 10852509      0.2470   0.0441530
#&gt; 8011:  rs9999767  chr4 10127270      0.8730   0.1902930</code></pre>
<p>Visualization of p-value distribution and comparison of the signals of GWAS and eQTL:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">xQTLvisual_locusCompare</span>(gwasEqtldata[,.(rsid, pValue.eqtl)], </span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>                        gwasEqtldata[,.(rsid, pValue.gwas)], <span class="at">legend_position =</span> <span class="st">&quot;bottomright&quot;</span>)</span></code></pre></div>
<p>Locuszoom plot of GWAS signals:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">xQTLvisual_locusZoom</span>(gwasEqtldata[,.(rsid, chrom, position, pValue.gwas)], <span class="at">legend=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p>Locuszoom plot of eQTL signals:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">xQTLvisual_locusZoom</span>(gwasEqtldata[,.(rsid, chrom, position, pValue.eqtl)], <span class="at">legend=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p>We can also combine locuscompare and locuszoom plot using function <code>xQTLvisual_locusCombine</code>:</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">xQTLvisual_locusCombine</span>(gwasEqtldata[,<span class="fu">c</span>(<span class="st">&quot;rsid&quot;</span>,<span class="st">&quot;chrom&quot;</span>, <span class="st">&quot;position&quot;</span>, <span class="st">&quot;pValue.gwas&quot;</span>, <span class="st">&quot;pValue.eqtl&quot;</span>)])</span></code></pre></div>
<p>From the above figures, we can see that the SNP <code>rs11568818</code> is potential causal variant, and we can use a violin plot to show the normalized effect size of it:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">xQTLvisual_eqtlExp</span>(<span class="st">&quot;rs11568818&quot;</span>, <span class="st">&quot;MMP7&quot;</span>, <span class="at">tissueSiteDetail =</span> tissueSiteDetail)</span></code></pre></div>
</div>
<div id="extension-integrative-analysis-with-external-packages-or-tools" class="section level2">
<h2>Extension: Integrative analysis with external packages or tools</h2>
<p>To gain insight into the function of trait genes with high PPH4, and explore the potential regulatory mechanism of the prostate cancer, we can conduct exploratory analysis, like co-expression analysis, and gene ontology enrichment analyses.</p>
<p>First we download expression profiles of the genes with higher value of PPH4 (&gt;0.75) in prostate.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>expMat <span class="ot">&lt;-</span> <span class="fu">xQTLdownload_exp</span>(outGenes<span class="sc">$</span>gencodeId, <span class="at">tissueSiteDetail=</span>tissueSiteDetail, <span class="at">toSummarizedExperiment =</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p>Pearson coefficient can be calculated with the expression matrix for each gene:</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>corDT <span class="ot">&lt;-</span> <span class="fu">cor</span>(<span class="fu">t</span>(expMat[,<span class="sc">-</span><span class="dv">1</span><span class="sc">:-</span><span class="dv">6</span>]))</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(corDT) <span class="ot">&lt;-</span> outGenes<span class="sc">$</span>geneSymbol</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(corDT) <span class="ot">&lt;-</span> outGenes<span class="sc">$</span>geneSymbol</span></code></pre></div>
<p>R package <code>corrplot</code> is used to display this correlation matrix:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(corrplot) </span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">corrplot</span>(corDT, <span class="at">method=</span><span class="st">&quot;color&quot;</span>,  </span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">type=</span><span class="st">&quot;upper&quot;</span>,    </span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">order =</span> <span class="st">&quot;hclust&quot;</span>,</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">addCoef.col =</span> <span class="st">&quot;#ff0099&quot;</span>, </span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">number.cex =</span> <span class="fl">0.7</span>) </span></code></pre></div>
<p>R package <code>clusterProfiler</code> is used for gene functional annotation:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> <span class="fu">as.character</span>(outGenes<span class="sc">$</span>entrezGeneId),</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">ont=</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(ego, <span class="at">showCategory=</span><span class="dv">15</span>)</span></code></pre></div>
<p>Viral-related GO terms are enriched in above analysis, including <code>&quot;viral life cycle&quot;</code>, <code>&quot;positive regulation of viral life cycle&quot;</code> and <code>&quot;positive regulation of viral process&quot;</code>. Previous studies have highlighten the role of viral infections in initiation or progression of prostate cancer. The presence of viruses such as human papillomavirus (HPV), herpesviruses including cytomegalovirus (CMV), human herpes simplex virus type 2 (HSV2), human herpesvirus type 8 (HHV8) and Epstein-Barr virus (EBV) can infect the prostate <span class="citation">(<a href="#ref-abidi2018" role="doc-biblioref">Abidi et al. 2018</a>)</span>. However, the causal variants’ genetic effects on the phenotype and whether the trait genes has a direct association with prostate carcinogenesis has not yet been established.</p>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="unnumbered">References</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-abidi2018" class="csl-entry">
Abidi, Syed Hani, Fareena Bilwani, Kulsoom Ghias, and Farhat Abbas. 2018. <span>“Viral etiology of prostate cancer: Genetic alterations and immune response. A literature review.”</span> <em>International Journal of Surgery (London, England)</em> 52 (April): 136–40. <a href="https://doi.org/10.1016/j.ijsu.2018.02.050">https://doi.org/10.1016/j.ijsu.2018.02.050</a>.
</div>
<div id="ref-RefSeq" class="csl-entry">
<span>“RefSeq.”</span> n.d. <a href="https://www.ncbi.nlm.nih.gov/gene/4316#summary">https://www.ncbi.nlm.nih.gov/gene/4316#summary</a>.
</div>
<div id="ref-tregunna2020" class="csl-entry">
Tregunna, Rebecca. 2020. <span>“Serum MMP7 levels could guide metastatic therapy for prostate cancer.”</span> <em>Nature Reviews. Urology</em> 17 (12): 658. <a href="https://doi.org/10.1038/s41585-020-00396-3">https://doi.org/10.1038/s41585-020-00396-3</a>.
</div>
<div id="ref-zhang2017" class="csl-entry">
Zhang, Q, S Liu, K R Parajuli, W Zhang, K Zhang, Z Mo, J Liu, et al. 2017. <span>“Interleukin-17 Promotes Prostate Cancer via Mmp7-Induced Epithelial-to-Mesenchymal Transition.”</span> <em>Oncogene</em> 36 (5): 687–99. <a href="https://doi.org/10.1038/onc.2016.240">https://doi.org/10.1038/onc.2016.240</a>.
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
