<!DOCTYPE html>

<html lang="en-US" xml:lang="en-US">

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />


<meta name="date" content="2023-05-01" />

<title>2. A quick application of coloclization analysis</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">2. A quick application of coloclization
analysis</h1>
<h4 class="date">2023-05-01</h4>



<p>xQTLbiolinks has provided a pipeline for disease target gene
discovery through colocalization analysis with xQTLs to detect
susceptible genes and causal variants.</p>
<div id="before-we-start" class="section level2">
<h2>Before we start</h2>
<p>To perform colocalization analysis using xQTLbiolinks, users need to
provide their own <strong>GWAS summary statistics data</strong>, specify
the <strong>tissue</strong> of interest and import following
packages:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(xQTLbiolinks)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(coloc)</span></code></pre></div>
</div>
<div id="step-1.-data-pre-processing" class="section level2">
<h2>Step 1. data pre-processing</h2>
<p>Download and load an example file of summary statistics dataset
(GRCh38). We perform colocalization analysis in
<code>Brain - Cerebellum</code>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>gwasDF <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;http://bioinfo.szbl.ac.cn/xQTL_biolinks/xqtl_data/gwasDFsub.txt&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>tissueSiteDetail<span class="ot">=</span><span class="st">&quot;Brain - Cerebellum&quot;</span></span></code></pre></div>
<p>In this example, a data.table object of 16538 (rows) x 5 (cols) is
loaded. Five columns are required (arbitrary column names is supported,
but columns must be as the following order):</p>
<p><code>Col 1</code>. “variants” (character), , using an rsID
(e.g. “rs11966562”);</p>
<p><code>Col 2</code>. “chromosome” (character), one of the chromosome
from chr1-chr22;</p>
<p><code>Col 3</code>. “position” (integer), genome position of snp;</p>
<p><code>Col 4</code>. “P-value” (numeric);</p>
<p><code>Col 5</code>. “MAF” (numeric). Allel frequency;</p>
<p><code>Col 6</code>. “beta” (numeric). effect size.</p>
<p><code>Col 7</code>. “se” (numeric). standard error.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">head</span>(gwasDF)</span></code></pre></div>
<pre><code>#&gt;          rsid chrom position   pValue     AF     beta      se
#&gt; 1: rs13120565     4 10702513 5.66e-10 0.6429  0.01825 0.00294
#&gt; 2:  rs4697781     4 10703243 8.94e-10 0.6446  0.01803 0.00294
#&gt; 3:  rs4697779     4 10701187 5.74e-09 0.3231 -0.01747 0.00300
#&gt; 4:  rs4697780     4 10701381 5.95e-09 0.3231 -0.01746 0.00300
#&gt; 5:  rs4547789     4 10702891 1.46e-08 0.3197 -0.01703 0.00300
#&gt; 6: rs11726285     4 10700944 1.47e-08 0.3214 -0.01702 0.00301</code></pre>
</div>
<div id="step-2.-identify-sentinel-snps." class="section level2">
<h2>Step 2. Identify sentinel snps.</h2>
<p>Sentinel SNPs can be detected using
<code>xQTLanalyze_getSentinelSnp</code> with the arguments
<code>p-value &lt; 5e-8</code> and
<code>SNP-to-SNP distance &gt; 10e6 bp</code>. We recommend converting
the genome version of the GWAS file to GRCh38 if that is GRCh37 (run
with parameter: <code>genomeVersion=&quot;grch37&quot;</code>;
<code>grch37To38=TRUE</code>, and package <code>rtracklayer</code>is
required).</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>sentinelSnpDF <span class="ot">&lt;-</span> <span class="fu">xQTLanalyze_getSentinelSnp</span>(gwasDF, <span class="at">pValueThreshold =</span> <span class="fl">5e-08</span>)</span></code></pre></div>
<p>After filtering, a sentinel SNP with P-value&lt;5e-8 is detected in
this example:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>sentinelSnpDF</span></code></pre></div>
<pre><code>#&gt;          rsid  chr position   pValue    maf    beta      se
#&gt; 1: rs13120565 chr4 10702513 5.66e-10 0.6429 0.01825 0.00294</code></pre>
</div>
<div id="step-3.-identify-trait-genes-for-each-sentinel-snps." class="section level2">
<h2>Step 3. Identify trait genes for each sentinel SNPs.</h2>
<p>Trait genes are genes that located in the range of 1Mb (default, can
be changed with parameter <code>detectRange</code>) of sentinel SNPs.
Every gene within 1Mb of sentinel SNPs is searched by fuction
<code>xQTLanalyze_getTraits</code>. Besides, In order to reduce the
number of trait genes and thus reduce the running time, we take the
overlap of eGenes and trait genes as the final output of the function
<code>xQTLanalyze_getTraits</code>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>traitsAll <span class="ot">&lt;-</span> <span class="fu">xQTLanalyze_getTraits</span>(sentinelSnpDF, <span class="at">detectRange=</span><span class="fl">1e6</span>, <span class="at">tissueSiteDetail=</span>tissueSiteDetail)</span></code></pre></div>
<p>Totally, 3 associations between 3 traits genes and 1 SNPs are
detected</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>traitsAll</span></code></pre></div>
<pre><code>#&gt;          rsid  chr position   pValue    maf    beta      se       gencodeId
#&gt; 1: rs13120565 chr4 10702513 5.66e-10 0.6429 0.01825 0.00294 ENSG00000002587
#&gt; 2: rs13120565 chr4 10702513 5.66e-10 0.6429 0.01825 0.00294 ENSG00000109684
#&gt; 3: rs13120565 chr4 10702513 5.66e-10 0.6429 0.01825 0.00294 ENSG00000261490</code></pre>
</div>
<div id="step-4.-conduct-colocalization-analysis." class="section level2">
<h2>Step 4. Conduct colocalization analysis.</h2>
<p><strong>For a single trait gene</strong>, like
<em>ENSG00000109684</em> in above table, colocalization analysis (using
<code>coloc</code> method) can be performed with:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>output <span class="ot">&lt;-</span> <span class="fu">xQTLanalyze_coloc</span>(gwasDF, <span class="st">&quot;ENSG00000109684&quot;</span>, <span class="at">tissueSiteDetail=</span>tissueSiteDetail) <span class="co"># using gene symbol</span></span></code></pre></div>
<p><em>output</em> is a list, including two parts:
<em>coloc_Out_summary</em> and <em>gwasEqtlInfo</em>.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>output<span class="sc">$</span>coloc_Out_summary</span></code></pre></div>
<pre><code>#&gt;    nsnps    PP.H0.abf   PP.H1.abf    PP.H2.abf  PP.H3.abf PP.H4.abf
#&gt; 1:  7107 7.097893e-11 1.32221e-07 3.890211e-06 0.00625302  0.993743
#&gt;          traitGene candidate_snp SNP.PP.H4
#&gt; 1: ENSG00000109684    rs13120565 0.5328849</code></pre>
<p><strong>For multiple trait genes</strong>, a for loop or lapply
function can be used to get all genes’ outputs (using both
<code>coloc</code> and <code>hyprcoloc</code> methods).</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>outputs <span class="ot">&lt;-</span> <span class="fu">rbindlist</span>(<span class="fu">lapply</span>( <span class="fu">unique</span>(traitsAll<span class="sc">$</span>gencodeId), <span class="cf">function</span>(x){ <span class="co"># using gencode ID.</span></span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>           <span class="fu">xQTLanalyze_coloc</span>(gwasDF, x, <span class="at">tissueSiteDetail=</span>tissueSiteDetail, <span class="at">method =</span> <span class="st">&quot;Both&quot;</span>)<span class="sc">$</span>colocOut }))</span></code></pre></div>
<p><em>outputs</em> is a data.table that combined all results of
<em>coloc_Out_summary</em> of all genes.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>outputs</span></code></pre></div>
<pre><code>#&gt;          traitGene nsnps    PP.H0.abf    PP.H1.abf    PP.H2.abf  PP.H3.abf
#&gt; 1: ENSG00000002587  6452 1.730175e-05 3.218430e-02 6.603361e-05 0.12198838
#&gt; 2: ENSG00000109684  7107 7.097893e-11 1.322210e-07 3.890211e-06 0.00625302
#&gt; 3: ENSG00000261490  6601 5.287051e-05 9.848309e-02 4.801374e-04 0.89435622
#&gt;     PP.H4.abf candidate_snp SNP.PP.H4 hypr_posterior hypr_regional_prob
#&gt; 1: 0.84574398    rs13120565 0.4140146         0.5685             0.9694
#&gt; 2: 0.99374296    rs13120565 0.5328849         0.9793             0.9999
#&gt; 3: 0.00662768    rs13120565 0.4219650         0.0000             0.0101
#&gt;    hypr_candidate_snp hypr_posterior_explainedBySnp
#&gt; 1:         rs13120565                        0.2726
#&gt; 2:         rs13120565                        0.4747
#&gt; 3:         rs13120565                        0.4118</code></pre>
</div>
<div id="step-5.-visualization-of-the-results." class="section level2">
<h2>Step 5. Visualization of the results.</h2>
<p>For the potential casual gene
<strong><em>ENSG00000109684</em></strong> (PP4=<strong>0.9937</strong>
&amp; hypr_posterior=<strong>0.9793</strong>), we can get its
significant associations across tissues:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="fu">xQTLvisual_eqtl</span>(<span class="st">&quot;ENSG00000109684&quot;</span>)</span></code></pre></div>
<p>For the purpose of visualization of p-value distribution and
comparison of the signals of GWAS and eQTL: We merge the variants of
GWAS and eQTL by rsid first.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>eqtlAsso <span class="ot">&lt;-</span> <span class="fu">xQTLdownload_eqtlAllAsso</span>(<span class="at">gene=</span><span class="st">&quot;ENSG00000109684&quot;</span>, </span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>                                     <span class="at">tissueLabel =</span> tissueSiteDetail)</span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>gwasEqtldata <span class="ot">&lt;-</span> <span class="fu">merge</span>(gwasDF, eqtlAsso[,.(<span class="at">rsid=</span>snpId, <span class="at">position=</span>pos, maf, pValue)],</span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>                      <span class="at">by=</span><span class="fu">c</span>(<span class="st">&quot;rsid&quot;</span>, <span class="st">&quot;position&quot;</span>), <span class="at">suffixes =</span> <span class="fu">c</span>(<span class="st">&quot;.gwas&quot;</span>,<span class="st">&quot;.eqtl&quot;</span>))</span></code></pre></div>
<p>Function <code>xQTLvisual_locusCompare</code> displays the candidate
SNP <strong><em>rs13120565</em></strong>
(SNP.PP.H4=<strong>0.5328849</strong> &amp;
hypr_posterior_explainedBySnp=<strong>0.4747</strong>) in the top right
corner:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">xQTLvisual_locusCompare</span>(gwasEqtldata[,.(rsid, pValue.eqtl)], </span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>                        gwasEqtldata[,.(rsid, pValue.gwas)], <span class="at">legend_position =</span> <span class="st">&quot;bottomright&quot;</span>)</span></code></pre></div>
<p><strong>Note: </strong>The information on SNP linkage disequilibrium
is automatically downloaded online. Due to network issues, the download
may fail at times. If this happens, please try running it again.</p>
<p>Locuszoom plot of GWAS signals:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="fu">xQTLvisual_locusZoom</span>(gwasEqtldata[,.(rsid, chrom, position, pValue.gwas)], <span class="at">legend=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p>LocusZoom plot of eQTL signals:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="fu">xQTLvisual_locusZoom</span>(gwasEqtldata[,.(rsid, chrom, position, pValue.eqtl)], </span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>                     <span class="at">highlightSnp =</span> <span class="st">&quot;rs13120565&quot;</span>, <span class="at">legend=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<p>Violin plot of normalized expression of eQTL
(rs13120565-ENSG00000187323.11):</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="fu">xQTLvisual_eqtlExp</span>(<span class="st">&quot;rs13120565&quot;</span>, <span class="st">&quot;ENSG00000109684&quot;</span>, <span class="at">tissueSiteDetail =</span> tissueSiteDetail)</span></code></pre></div>
<p>We can also combine locuscompare and locuszoom plot using
<code>xQTLvisual_locusCombine</code>:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="fu">xQTLvisual_locusCombine</span>(gwasEqtldata[,<span class="fu">c</span>(<span class="st">&quot;rsid&quot;</span>,<span class="st">&quot;chrom&quot;</span>, <span class="st">&quot;position&quot;</span>, <span class="st">&quot;pValue.gwas&quot;</span>, <span class="st">&quot;pValue.eqtl&quot;</span>)], </span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>                        <span class="at">highlightSnp=</span><span class="st">&quot;rs13120565&quot;</span>)</span></code></pre></div>
<p>Colocalized loci should show a general pattern where SNPs in high LD
will show strong associations with expression levels of the colocalized
gene, and the eQTL associations will weaken for SNPs in lower LD. This
pattern of the eQTL varies among different tissues / cell-types, the
strength of which indicates the power of the regulatory effect of the
variant. We can visualize the correlation between p-values of eQTLs and
LD across numerous tissues/cell types to easily distinguish this patten
using <code>xQTLvisual_coloc</code>:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>multi_tissue_coloc <span class="ot">&lt;-</span> <span class="fu">xQTLvisual_coloc</span>(<span class="at">gene=</span><span class="st">&quot;ENSG00000109684&quot;</span>, <span class="at">variantName=</span><span class="st">&quot;rs13120565&quot;</span>, </span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>                                      <span class="at">tissueLabels =</span> <span class="fu">c</span>(<span class="st">&quot;Brain - Cerebellar Hemisphere&quot;</span>, </span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>                                                        <span class="st">&quot;Brain - Cerebellum&quot;</span>, <span class="st">&quot;Thyroid&quot;</span>, <span class="st">&quot;Lung&quot;</span>,</span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>                                                        <span class="st">&quot;Cells - EBV-transformed lymphocytes&quot;</span>))</span></code></pre></div>
<p>For applying <code>xQTLbiolinks</code> to a whole case study, please
find this <a href="https://dingruofan.github.io/xQTLbiolinks/articles/Colocalization_analysis_with_xQTLbiolinks.html"><strong>Document</strong></a>.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
